airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

```diff
@@ -0,0 +1,150 @@
+        (IdP) that has been defined for your account
+        'https://<your_okta_account_name>.okta.com' to authenticate
+        through native Okta.
+    :type authenticator: str
+    :param session_parameters: You can set session-level parameters at
+        the time you connect to Snowflake
+    :type session_parameters: dict
+    """
+
+    template_fields = (
+        's3_key',
+        's3_bucket',
+        'table_or_query',
+        'snowflake_conn_id',
+    )
+    template_ext = ('.sql',)
+    ui_color = '#ffebb2'
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        stage: Optional[str] = None,
+        s3_bucket: Optional[str],
+        s3_key: str,
+        query_or_table: str,
+        file_format: Optional[str],
+        database: Optional[str] = None,
+        warehouse: Optional[str] = None,
+        snowflake_conn_id: str = "snowflake_default",
+        unload_options: Optional[list] = None,
+        autocommit: bool = True,
+        include_header: bool = False,
+        role: Optional[str] = None,
+        authenticator: Optional[str] = None,
+        session_parameters: Optional[dict] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.stage = stage
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+        self.file_format = file_format
+        self.warehouse = warehouse
+        self.database = database
+        self.snowflake_conn_id = snowflake_conn_id
+        self.query_or_table = query_or_table
+        self.unload_options = unload_options or []  # type: List
+        self.autocommit = autocommit
+        self.include_header = include_header
+        self.role = role
+        self.authenticator = authenticator
+        self.session_parameters = session_parameters
+
+    def execute(self, context):
+        snowflake_hook = SnowflakeHook(
+            snowflake_conn_id=self.snowflake_conn_id,
+            warehouse=self.warehouse,
+            database=self.database,
+            role=self.role,
+            authenticator=self.authenticator,
+            session_parameters=self.session_parameters,
+        )
+        unload_options = '\n\t'.join(self.unload_options)
+
+        if self.stage:
+            unload_query = f"""
+            COPY INTO @{self.stage}/{self.s3_key}
+            FROM ({self.query_or_table})
+            """
+
+            if self.file_format:
+                unload_query += f'FILE_FORMAT = ({self.file_format})'
+
+        else:
+            unload_query = f"""
+            COPY INTO 's3://{self.s3_bucket}/{self.s3_key}'
+            FROM ({self.query_or_table})
+            STORAGE_INTEGRATION = S3
+            FILE_FORMAT = ({self.file_format})
+            """
+
+        unload_query += f"""{unload_options}
+                            HEADER = {self.include_header};"""
+
+        self.log.info('Executing UNLOAD command...')
+        snowflake_hook.run(unload_query, self.autocommit)
```

Comment:
i recommend using cur.execute instead so you can run cur.fetchall afterwards and print to logs the output 

snowflake tells you the files it created in the unload, and this is useful logging information.

same as i suggested with the snowflake load scenario

Reply:
Do you think it makes sense to include the cur.fetchall directly in the hook? 

Also, I recall reading somewhere that it is not necessary to open the cursor like a context because snowflake handles that very well, but I don't find it. Do you know anything about it? Thanks!

Reply:
> Do you think it makes sense to include the cur.fetchall directly in the hook?

yes absolutely.  ideally you should also parse the results and provide configurable failure behavior.  e.g. should the task fail if any files are skipped? or if any files are failed? 

> Also, I recall reading somewhere that it is not necessary to open the cursor like a context because snowflake handles that very well, but I don't find it. Do you know anything about it? Thanks!

That might be true.  However, all of the example code in snowflake docs demonstrate closing of connections, and here they mention explicitly that it should be done:

https://docs.snowflake.com/en/user-guide/python-connector-example.html#closing-the-connection